spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -48,145 +48,150 @@ public void testV2Filters() {
+
+    Predicate lt2 = new Predicate("<", valueAndAttr);
+    Expression expectedLt2 = Expressions.greaterThan(unquoted, 1);
+    Expression actualLt2 = SparkV2Filters.convert(lt2);
+    Assert.assertEquals("LessThan must match", expectedLt2.toString(), actualLt2.toString());
+
+    Predicate ltEq1 = new Predicate("<=", attrAndValue);
+    Expression expectedLtEq1 = Expressions.lessThanOrEqual(unquoted, 1);
+    Expression actualLtEq1 = SparkV2Filters.convert(ltEq1);
+    Assert.assertEquals(
+        "LessThanOrEqual must match", expectedLtEq1.toString(), actualLtEq1.toString());
+
+    Predicate ltEq2 = new Predicate("<=", valueAndAttr);
+    Expression expectedLtEq2 = Expressions.greaterThanOrEqual(unquoted, 1);
+    Expression actualLtEq2 = SparkV2Filters.convert(ltEq2);
+    Assert.assertEquals(
+        "LessThanOrEqual must match", expectedLtEq2.toString(), actualLtEq2.toString());
+
+    Predicate gt1 = new Predicate(">", attrAndValue);
+    Expression expectedGt1 = Expressions.greaterThan(unquoted, 1);
+    Expression actualGt1 = SparkV2Filters.convert(gt1);
+    Assert.assertEquals("GreaterThan must match", expectedGt1.toString(), actualGt1.toString());
+
+    Predicate gt2 = new Predicate(">", valueAndAttr);
+    Expression expectedGt2 = Expressions.lessThan(unquoted, 1);
+    Expression actualGt2 = SparkV2Filters.convert(gt2);
+    Assert.assertEquals("GreaterThan must match", expectedGt2.toString(), actualGt2.toString());
+
+    Predicate gtEq1 = new Predicate(">=", attrAndValue);
+    Expression expectedGtEq1 = Expressions.greaterThanOrEqual(unquoted, 1);
+    Expression actualGtEq1 = SparkV2Filters.convert(gtEq1);
+    Assert.assertEquals(
+        "GreaterThanOrEqual must match", expectedGtEq1.toString(), actualGtEq1.toString());
+
+    Predicate gtEq2 = new Predicate(">=", valueAndAttr);
+    Expression expectedGtEq2 = Expressions.lessThanOrEqual(unquoted, 1);
+    Expression actualGtEq2 = SparkV2Filters.convert(gtEq2);
+    Assert.assertEquals(
+        "GreaterThanOrEqual must match", expectedGtEq2.toString(), actualGtEq2.toString());
+
+    Predicate eq1 = new Predicate("=", attrAndValue);
+    Expression expectedEq1 = Expressions.equal(unquoted, 1);
+    Expression actualEq1 = SparkV2Filters.convert(eq1);
+    Assert.assertEquals("EqualTo must match", expectedEq1.toString(), actualEq1.toString());
+
+    Predicate eq2 = new Predicate("=", valueAndAttr);
+    Expression expectedEq2 = Expressions.equal(unquoted, 1);
+    Expression actualEq2 = SparkV2Filters.convert(eq2);
+    Assert.assertEquals("EqualTo must match", expectedEq2.toString(), actualEq2.toString());
+
+    Predicate notEq1 = new Predicate("<>", attrAndValue);
+    Expression expectedNotEq1 = Expressions.notEqual(unquoted, 1);
+    Expression actualNotEq1 = SparkV2Filters.convert(notEq1);
+    Assert.assertEquals(
+        "NotEqualTo must match", expectedNotEq1.toString(), actualNotEq1.toString());
+
+    Predicate notEq2 = new Predicate("<>", valueAndAttr);
```

Comment:
Are there cases for null and NaN values? I think there should be since there is special handling for them.

Reply:
Added